scrapi/harvesters/noaa_nodc.py

```diff
@@ -0,0 +1,226 @@
+        }
+
+
+def filter_keywords(keyword_groups):
+    '''Filter out the NODC org name from the keywords to reduce noise
+    and flatten nested lists'''
+    keywords = []
+    for keyword_group in keyword_groups:
+        keyword_type = keyword_group.xpath('./gmd:type/gmd:MD_KeywordTypeCode/node()', namespaces=keyword_group.nsmap)
+        if len(keyword_type) > 0 and keyword_type[0] != 'datacenter':
+            keywords.extend(keyword_group.xpath('./gmd:keyword/gmx:Anchor/node()', namespaces=keyword_group.nsmap))
+    return keywords
+
+
+class NODCHarvester(XMLHarvester):
+    short_name = 'noaa_nodc'
+    long_name = 'National Oceanographic Data Center'
+    url = 'https://www.nodc.noaa.gov/'
+
+    DEFAULT_ENCODING = 'UTF-8'
+    record_encoding = None
+
+    canonical_base_url = 'http://data.nodc.noaa.gov/cgi-bin/iso?id={}'
+
+    namespaces = {
+        'gco': 'http://www.isotc211.org/2005/gco',
+        'gmd': 'http://www.isotc211.org/2005/gmd',
+        'gmi': 'http://www.isotc211.org/2005/gmi',
+        'gml': 'http://www.opengis.net/gml/3.2',
+        'gmx': 'http://www.isotc211.org/2005/gmx',
+        'gsr': 'http://www.isotc211.org/2005/gsr',
+        'gss': 'http://www.isotc211.org/2005/gss',
+        'gts': 'http://www.isotc211.org/2005/gts',
+        'srv': 'http://www.isotc211.org/2005/srv',
+        'xlink': 'http://www.w3.org/1999/xlink',
+        'xsi': 'http://www.w3.org/2001/XMLSchema-instance',
+        # xsi:schemaLocation="http://www.ngdc.noaa.gov/metadata/published/xsd/schema.xsd",
+    }
+
+    @property
+    def schema(self):
+        id_stanza = './gmd:identificationInfo/gmd:MD_DataIdentification/'
+        cite_stanza = id_stanza + 'gmd:citation/gmd:CI_Citation/'
+        return {
+            'title': (cite_stanza + 'gmd:title', compose(xml_text_only, single_result)),
+            'description': (id_stanza + 'gmd:abstract', compose(xml_text_only, single_result)),
+            'contributors': (cite_stanza + 'gmd:citedResponsibleParty/gmd:CI_ResponsibleParty', compose(parse_contributors, filter_to_contributors)),
+            'uris': {
+                'canonicalUri': (
+                    './gmd:fileIdentifier',
+                    compose(lambda x: str(self.canonical_base_url).format(x), xml_text_only, single_result)
+                ),
+            },
+            'publisher': (
+                cite_stanza + 'gmd:citedResponsibleParty/gmd:CI_ResponsibleParty',
+                compose(extract_organization, single_result, filter_to_publishers),
+            ),
+            'providerUpdatedDateTime': ('./gmd:dateStamp/gco:DateTime/node()', compose(date_formatter, single_result)),
+            'languages': ('./gmd:language/gmd:LanguageCode', compose(language_codes, xml_text_only_list, coerce_to_list)),
+            'subjects': (id_stanza + 'gmd:descriptiveKeywords/gmd:MD_Keywords', lambda x: filter_keywords(x)),
+        }
+
+    search_base_url = 'http://data.nodc.noaa.gov/cgi-bin/oai-pmh?verb=ListRecords'
```

Comment:
I would put these up at the top of the class definition with the rest of the definitions.